### ISSUE TITLE
"undo"ing an error doesn't cause sign to reappear.

### ISSUE THREAD

**Issue by sbromberger** [haorenW1025/diagnostic-nvim#16]
If you introduce an error, the sign comes up in the gutter. If you then delete the line containing the error, the sign disappears (yay!). However, if you `u`ndo the deletion, causing the error to reappear, the sign doesn't reappear.

here's a demo:
https://asciinema.org/a/R2WqJOXIzoxW4e3QZqRA3w2Ku

**Comment by haorenW1025** [comment]
Hmm have you tried tested it without this plugin? I tried `clangd`, `rust_analyzer` and  `sumneko_lua` and both of the server still produce this error without `diagnostic-nvim`.

**Comment by sbromberger** [comment]
Sorry. *I* didn't but someone on the neovim matrix channel reported that it didn't affect them (no diagnostic-nvim plugin).

**Comment by sbromberger** [comment]
this was with rls, btw.

**Comment by haorenW1025** [comment]
@clason can I ask what lsp you've used to test this issue? Thanks!

**Comment by clason** [comment]
I'm using https://github.com/latex-lsp/texlab, here on a bib file:
```bibtex
@article{test,
title = {A title}
}
```
this should show an error in the second line (missing terminating comma).

**Comment by haorenW1025** [comment]
Did some testing with `texlab` it turns out the sign will reappear even when `diagnostic-nvim` is on. So I'm starting to think that this should be an issue with the server itself...

![Kapture 2020-05-04 at 22 03 12](https://user-images.githubusercontent.com/35623968/80974375-3886a780-8e53-11ea-823f-3104b417ebbd.gif)

**Comment by sbromberger** [comment]
Sorry for the false alarm. Unfortunately, since `rls` is being deprecated in favor of `rust-analyzer` soon, my guess is that this won't be fixed.

It's too bad. `rust-analyzer` doesn't do dynamic linting, so it's not even remotely a practical replacement for many people.
